registrations/tasks.py

```diff
@@ -43,3 +44,41 @@ def send_registration_to_openhim(
     )
     response.raise_for_status()
     return (response.status_code, response.headers, response.content)
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded),
+    retry_backoff=True,
+    max_retries=15,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def send_registration_to_rapidpro(
+    contact, msisdn, referral_msisdn, channel, clinic_code, timestamp
+):
+    # Create/Update contact
+    group_name = "nurseconnect-%s" % channel.lower()
+    group = tembaclient.get_groups(name=group_name).first()
+    contact_data = {
+        "preferred_channel": channel.lower(),
+        "registered_by": referral_msisdn,
+        "facility_code": clinic_code,
+        "registration_date": datetime.fromtimestamp(timestamp).isoformat(),
```

Comment:
The RapidPro client automatically serialises this for us, so we can just pass in the `datetime` object.

Reply:
@rudigiesler when I change this to just `datetime.fromtimestamp(timestamp)` then I get
`Object of type 'datetime' is not JSON serializable`

Reply:
Oh that's really weird... If I'm reading their code correctly, it should serialize it: https://github.com/rapidpro/rapidpro-python/blob/master/temba_client/clients.py#L141-L160 . Maybe we can use their serializer then?

Reply:
Ah wait, I see the issue here. What's being send through to the serializer is `fields={.."registration_date": ...}`, so it's not checking if the registration date is a datetime, it's checking if the whole fields dictionary is a datetime.

Reply:
So I think we should either use their serializer, or not using `isoformat`, since the jsbox serializer is a very specific value, that is different from what `isoformat` gives us: https://momentjs.com/docs/#/displaying/format/

Reply:
ok yeah, `TembaClient._serialize_value(datetime.fromtimestamp(timestamp))` works.
And solves the problem of it not showing up in the UI.
Unfortunate that it's a private method :/

Reply:
:confused: that says "moment().format();  // "2014-09-08T08:02:17-05:00" (ISO 8601, no fractional seconds)" and the isoformat docs say "Return a string representing the date and time in ISO 8601 format"
but I guess it's the microseconds that are causing the discrepancy?
Anyway, I think I'll use their serialiser. That way we're less likely to encounter problems if they change the format.

Reply:
Since we know it's a datetime, we can probably use the `utils.format_iso8601` directly, which isn't a private method.